bundles/org.openhab.binding.linky/src/main/java/org/openhab/binding/linky/internal/LinkyHandlerFactory.java

```diff
@@ -107,12 +154,141 @@ protected void deactivate(ComponentContext componentContext) {
 
     @Override
     public boolean supportsThingType(ThingTypeUID thingTypeUID) {
-        return THING_TYPE_LINKY.equals(thingTypeUID);
+        return LinkyBindingConstants.THING_TYPE_LINKY.equals(thingTypeUID);
     }
 
     @Override
     protected @Nullable ThingHandler createHandler(Thing thing) {
-        return supportsThingType(thing.getThingTypeUID()) ? new LinkyHandler(thing, localeProvider, gson, httpClient)
-                : null;
+
+        if (supportsThingType(thing.getThingTypeUID())) {
+            LinkyHandler handler = new LinkyHandler(thing, localeProvider, gson, httpClient);
+            return handler;
+        }
+
+        return null;
+    }
+
+    // ===========================================================================
```

Comment:
All below lines probably has to be moved in a distinct class.

Reply:
I refactor LinkyHandlerFactory, moving oAuth stuff to LinkyHandler

Reply:
Sorry, this is move to ApiBridgeHandler, not LinkyHandler !